packages/ec-core/src/operator/selector/tournament.rs

```diff
@@ -1,18 +1,60 @@
+use std::num::NonZeroUsize;
+
 use anyhow::{ensure, Context, Result};
 use rand::{prelude::IndexedRandom, rngs::ThreadRng};
 
 use super::Selector;
 use crate::population::Population;
 
 pub struct Tournament {
-    size: usize,
+    size: NonZeroUsize,
 }
 
 impl Tournament {
+    /// Construct a tournament selector with the given tournament size. This
+    /// will select `size` individuals from the population, randomly
+    /// without replacement, and return the "best" from that set.
     #[must_use]
-    pub const fn new(size: usize) -> Self {
+    pub const fn new(size: NonZeroUsize) -> Self {
         Self { size }
     }
+
+    /// Construct a tournament selector with the given _constant_
+    /// tournament size, allowing for compile-time checks that the
+    /// tournament size `N` is greater than zero. This selector
+    /// will select `size` individuals from the population, randomly
+    /// without replacement, and return the "best" from that set.
+    ///
+    /// # Examples
+    ///
+    /// Create a tournament selector with tournament size 3:
+    /// ```
+    /// # use ec_core::operator::selector::tournament::Tournament;
+    /// let selector = Tournament::of_size::<3>();
+    /// ```
+    ///
+    /// # Panics
```

Comment:
Actually this is kind of misleading since this can *never* panic at runtime, only during compile-time, which will then show a error during compilation - maybe the clippy `clippy::missing_panics_doc` lint doesn't play super nice here? I would definitely at least note that here that this is not runtime behaviour